src/pkg/utils/network.go

```diff
@@ -70,26 +69,52 @@ func Fetch(url string) io.ReadCloser {
 }
 
 // DownloadToFile downloads a given URL to the target filepath (including the cosign key if necessary).
-func DownloadToFile(url string, target string, cosignKeyPath string) {
-
-	// Always ensure the target directory exists
-	if err := CreateFilePath(target); err != nil {
-		message.Fatalf(err, "Unable to create file path: %s", target)
+func DownloadToFile(src string, dst string, cosignKeyPath string) (err error) {
+	parsed, err := url.Parse(src)
+	if err != nil {
+		return fmt.Errorf("unable to parse the URL: %s", src)
+	}
+	// check if the parsed URL has a checksum
+	// if so, remove it and use the checksum to validate the file
+	hasChecksum := strings.Contains(parsed.Path, "@")
+	var checksum string
+	if hasChecksum {
+		index := strings.LastIndex(parsed.Path, "@")
```

Comment:
Pretty sure I could break this with a URL like:

```
https://example.com/manifest?type=yaml@348217fbca997689e9f98978ab
```

Reply:
now fixed in unit tests